src/pveclib/vec_f128_ppc.h

```diff
@@ -4286,64 +5033,251 @@ vec_cmpqp_all_ne (__binary128 vfa, __binary128 vfb)
+ *  operation. This leverages operations from vec_int128_ppc.h.
+ *
+ *  \note This operation <I>may not</I> follow the PowerISA
+ *  relative to setting the FPSCR.
+ *  However if the hardware target includes an instruction that does
+ *  implement the IEEE standard, the implementation may use that.
+ *  This relaxed implementation may be useful for implementations on
+ *  POWER8 and earlier. Especially for soft-float implementations
+ *  where it is known these special cases do not occur.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   |  8-17 | 1/cycle  |
+ *  |power9   |   3   | 2/cycle  |
+ *
+ *  @param vfa 128-bit vector treated as a scalar __binary128.
+ *  @param vfb 128-bit vector treated as a scalar __binary128.
+ *  @return int boolean reflecting __binary128 unordered.
+ */
+
+static inline int
+vec_cmpqp_exp_unordered (__binary128 vfa, __binary128 vfb)
+{
+#if defined (_ARCH_PWR9) && defined (scalar_cmp_exp_gt) \
+  && defined (__FLOAT128__) && (__GNUC__ >= 9)
+  return scalar_cmp_exp_unordered (vfa, vfb);
+#else
+  vui32_t vra, vrb;
+  const vui32_t expmask = CONST_VINT128_W(0x7fff0000, 0, 0, 0);
+
+  vra = vec_and_bin128_2_vui32t (vfa, expmask);
+  vrb = vec_and_bin128_2_vui32t (vfb, expmask);
+  return vec_cmpuq_all_lt ((vui128_t) vra, (vui128_t) vrb);
+#endif
+}
+
+/** \brief Return 128-bit vector boolean true if the __float128 value
+ *  is Finite (Not NaN nor Inf).
+ *
+ *  A IEEE Binary128 finite value has an exponent between 0x0000 and
+ *  0x7ffe (a 0x7fff indicates NaN or Inf).  The significand can be
+ *  any value.
+ *  Using the vec_cmpeq conditional to generate the predicate mask for
+ *  NaN / Inf and then invert this for the finite condition.
+ *  The sign bit is ignored.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   | 8-17  | 2/cycle  |
+ *  |power9   |   6   | 2/cycle  |
+ *
+ *  \note This function will not raise VXSNAN or VXVC (FE_INVALID)
+ *  exceptions. A normal __float128 compare can.
+ *
+ *  @param f128 a __float128 value in vector.
+ *  @return a vector boolean containing all 0s or 1s.
+ */
+static inline vb128_t
+vec_isfinitef128 (__binary128 f128)
+{
+#if defined (_ARCH_PWR9) && defined (scalar_test_data_class) && defined (__FLOAT128__) && (__GNUC__ > 7)
+  vui32_t result = CONST_VINT128_W(-1, -1, -1, -1);
+
+  if (scalar_test_data_class (f128, 0x70))
+    result = CONST_VINT128_W(0, 0, 0, 0);
+
+  return (vb128_t)result;
+#else
+  const vui32_t expmask = CONST_VINT128_W(0x7fff0000, 0, 0, 0);
+  vui32_t tmp;
+  vb128_t tmp2, tmp3;
+
+  tmp = vec_and_bin128_2_vui32t (f128, expmask);
+  tmp2 = (vb128_t) vec_cmpeq (tmp, expmask);
+  tmp3 = (vb128_t) vec_splat ((vui32_t) tmp2, VEC_W_H);
+  return (vb128_t) vec_nor ((vui32_t) tmp3, (vui32_t) tmp3); // vec_not
+#endif
+}
+
+/** \brief Return true (nonzero) value if the __float128 value is
+ * infinity. For infinity indicate the sign as +1 for positive infinity
```

Comment:
comma after "For infinity"

Reply:
I think ..
/If infinity, indicate the sign as/